src/main/java/org/nuxeo/audit/storage/impl/SQLAuditStorage.java

```diff
@@ -0,0 +1,203 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Estelle Giuly <egiuly@nuxeo.com>
+ */
+
+package org.nuxeo.audit.storage.impl;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.codehaus.jackson.annotate.JsonProperty;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.nuxeo.ecm.core.api.CursorService;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.ScrollResult;
+import org.nuxeo.ecm.core.query.sql.model.MultiExpression;
+import org.nuxeo.ecm.core.query.sql.model.Predicate;
+import org.nuxeo.ecm.core.query.sql.model.StringLiteral;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+import org.nuxeo.ecm.directory.sql.SQLDirectory;
+import org.nuxeo.ecm.directory.sql.SQLSession;
+import org.nuxeo.ecm.platform.audit.api.AuditQueryBuilder;
+import org.nuxeo.ecm.platform.audit.api.AuditStorage;
+import org.nuxeo.ecm.platform.audit.api.ExtendedInfo;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.audit.impl.LogEntryImpl;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Audit storage implementation for a SQL Audit directory.
+ * 
+ * @since 9.3
+ */
+public class SQLAuditStorage implements AuditStorage {
+
+    private static final Log log = LogFactory.getLog(SQLAuditStorage.class);
+
+    public static final String NAME = "SQLAuditStorage";
+
+    public static final String DIRECTORY_NAME = "sqlAuditDirectory";
+
+    public static final String ID_COLUMN = "id";
+
+    public static final String JSON_COLUMN = "entry";
+
+    protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    static {
+        OBJECT_MAPPER.getDeserializationConfig().addMixInAnnotations(LogEntryImpl.class, LogEntryImplMixIn.class);
+    }
+
+    protected CursorService<Iterator<LogEntry>, LogEntry> cursorService = new CursorService<>();
+
+    protected SQLDirectory sqlDirectory;
+
+    public SQLDirectory getSqlDirectory() {
```

Comment:
`protected` if possible. If you need to keep it `public` and it's for tests, note it with `// public for tests` but I don't think that's the case.